test/performance/operf.go

```diff
@@ -0,0 +1,538 @@
+
+type metric struct {
+	Max float64
+	Avg float64
+}
+
+type lvmUnitMetrics struct {
+	Namespace    string
+	Name         string
+	WorkloadType string
+	cpu          metric
+	ram          metric
+	fs           metric
+	start        time.Time
+	end          time.Time
+}
+
+const (
+	cpuQueryTemplate   = `sum(node_namespace_pod_container:container_cpu_usage_seconds_total:sum_irate{cluster="", namespace="{{.Namespace}}"} * on(namespace,pod) group_left(workload, workload_type) namespace_workload_pod:kube_pod_owner:relabel{cluster="", namespace="{{.Namespace}}", workload="{{.Name}}", workload_type="{{.WorkloadType}}"}) by (pod)`
+	memQueryTemplate   = `sum(container_memory_working_set_bytes{cluster="", namespace="{{.Namespace}}",container!="", image!=""} * on(namespace,pod) group_left(workload, workload_type) namespace_workload_pod:kube_pod_owner:relabel{cluster="", namespace="{{.Namespace}}", workload="{{.Name}}", workload_type="{{.WorkloadType}}"}) by (pod)`
+	filesystemTemplate = `sum(pod:container_fs_usage_bytes:sum * on(pod) group_left(workload, workload_type) namespace_workload_pod:kube_pod_owner:relabel{workload="{{.Name}}", workload_type="{{.WorkloadType}}"}) by (pod)`
+	Mebibyte           = 1048576 // bytes
+)
+
+const (
+	usageHelp = `Usage of operf:
+This script provides LVM operator metrics retrieval (for all the units in the operator):
+
+- PVCs creation and usage:
+	It creates <instances> of "busybox" pods using PVCs provisioned via the Storage class provided
+	Example:
+	# go run test/performance/operf.go -token sha256~cj81ClyUYu7g05y8K-uLWm2AbrKTbNEQ96hEJcWStQo -instances 16
+
+Parameters:`
+)
+
+var (
+	// flags
+	instances            *int
+	testNamespace        *string
+	testStorageClassName *string
+	namePattern          *string
+	token                *string
+
+	// variables
+	PVCLabels     = map[string]string{"app": "testPerformance"}
+	prometheusURL string
+)
+
+func main() {
+	instances = flag.Int("instances", 4, "Number of Pods/Pvcs (each Pod uses a different PVC) to create in the test")
+	testNamespace = flag.String("namespace", "lvm-operator-system", "Namespace where operator is deployed and the PVCs and test pods will be deployed/undeployed")
+	testStorageClassName = flag.String("sc", "odf-lvm-vg1", "Name of the topolvm storage class that will be used in the PVCs")
+	namePattern = flag.String("pattern", "perfotest", "Pattern used to build the PVCs/Pods names")
+	token = flag.String("token", "", "Mandatory authentication token needed to connect with the Openshift cluster")
+
+	flag.Usage = func() {
+		writer := tabwriter.NewWriter(os.Stdout, 0, 8, 1, '\t', tabwriter.AlignRight)
+		flagSet := flag.CommandLine
+		fmt.Println(usageHelp)
+		order := []string{"instances", "namespace", "pattern", "sc", "token"}
+		for _, name := range order {
+			flag := flagSet.Lookup(name)
+			fmt.Fprintf(writer, "\t-%s\t(default: %s)\t%s\n", flag.Name, flag.DefValue, flag.Usage)
+		}
+		writer.Flush()
+	}
+
+	flag.Parse()
+
+	if *token == "" {
```

Comment:
If the token is not passed, will it panic with null pointer exception?

Reply:
It was just aborting test via return, Will panic now